### DISCUSSION TITLE
How do I drive a 512 bit input to my DUT

### DISCUSSION THREAD

**Discussion by ghost** [cocotb/cocotb#3279] [Q&A]
Hi, 
Is there a way to drive a 512 bit input to my DUT?

I tried creating a hex string and converting it to binary to use `BinaryValue`
```
myval = b'AABBCCDD....' 
dut.mysig.value=BinaryValue(value=bin(int(myval, 16)), n_bits=512)
```

but ran into issues. Is there a recommended way to drive this input?

**Comment by silver-signal** [comment]
My preference is to pass values in using the buff param of BinaryValue, but the best method may depend on your application and the form that other python functions may expect.

In your code, a string of binary is being passed to 'value', which is treated as an integer (and is deprecated).

This would likely fix your issue:

```
my_val = b'AABBCCDD'
my_binary = bin(int(my_val, 16))
my_binary = my_binary[2:] # Remove the '0b' from the front of the string

vec = BinaryValue()
vec.binstr = my_binary

# Show that the values are what we'd expect
print(vec.binstr) # '10101010101110111100110011011101'
print(vec.integer) # 2864434397
print(vec.buff) # b'\xaa\xbb\xcc\xdd' <- These bytes should match 'my_val'
```

Or a more direct way with the python binascii stdlib:

```
from binascii import unhexlify

my_val = b'AABBCCDD'

vec = BinaryValue()
vec.buff = unhexlify(my_val)

# Show that the values are what we'd expect
print(vec.binstr) # '10101010101110111100110011011101'
print(vec.integer) # 2864434397
print(vec.buff) # b'\xaa\xbb\xcc\xdd' <- These bytes should match 'my_val'
```
